x-pack/plugins/infra/public/pages/metrics/hosts/components/metrics_trend/metrics_chart.tsx

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useMemo, useCallback } from 'react';
+import {
+  Chart,
+  Metric,
+  MetricTrendShape,
+  type MetricWNumber,
+  type MetricWTrend,
+} from '@elastic/charts';
+
+import { EuiPanel } from '@elastic/eui';
+import styled from 'styled-components';
+import { EuiLoadingChart } from '@elastic/eui';
+import { EuiFlexGroup } from '@elastic/eui';
+import { EuiFlexItem } from '@elastic/eui';
+import { EuiToolTip } from '@elastic/eui';
+import type { SnapshotNode, SnapshotNodeMetric } from '../../../../../../common/http_api';
+import { createInventoryMetricFormatter } from '../../../inventory_view/lib/create_inventory_metric_formatter';
+import type { SnapshotMetricType } from '../../../../../../common/inventory_models/types';
+
+type MetricType = keyof Pick<SnapshotNodeMetric, 'avg' | 'max' | 'value'>;
+
+type AcceptedType = SnapshotMetricType | 'hostsCount';
+
+export interface ChartBaseProps
+  extends Pick<
+    MetricWTrend,
+    'title' | 'color' | 'extra' | 'subtitle' | 'trendA11yDescription' | 'trendA11yTitle'
+  > {
+  type: AcceptedType;
+  toolTip: string;
+  metricType: MetricType;
+  ['data-test-subj']?: string;
+}
+
+interface Props extends ChartBaseProps {
+  id: string;
+  nodes: SnapshotNode[];
+  loading: boolean;
+  overrideValue?: number;
+}
+
+const MIN_HEIGHT = 150;
+
+export const MetricsChart = ({
+  color,
+  extra,
+  id,
+  loading,
+  metricType,
+  nodes,
+  overrideValue,
+  subtitle,
+  title,
+  toolTip,
+  trendA11yDescription,
+  trendA11yTitle,
+  type,
+  ...props
+}: Props) => {
+  const metrics = useMemo(() => (nodes ?? [])[0]?.metrics ?? [], [nodes]);
+
+  const getTimeseries = useCallback(
```

Comment:
If `getTimeseries` and `getValue` callbacks are memoized only to be used inside the useMemo computation for `metricsTimeseries` and `metricsValue`, you can directly make these computations inside the useMemo function, without adding two useCallback steps in the hooks flow for this component
```ts
const metricsTimeseries = useMemo(() => {
  if (!metrics || !metrics.length) {
    return null;
  }
  return metrics.find((m) => m.name === type)?.timeseries;
}, [metrics, type]);
```

The same happens to `metricsValue`

Reply:
Very true. Can't remember why I did that 😕